src/modules/ticket-fields/data-types/TicketFieldObject.ts

```diff
@@ -0,0 +1,21 @@
+import type { BaseTicketField } from "./BaseTicketField";
+
+/**
+ * This interface represents the
+ * [ticket field object](https://developer.zendesk.com/api-reference/help_center/help-center-templates/objects/#ticket-field-object)
+ * available in the New Request Page template.
+ */
+export interface TicketFieldObject extends BaseTicketField {
+  id: number;
+  name: string;
+  value: string | string[] | boolean | null;
+  error: string | null;
+  label: string;
+  required: boolean;
+  options: FieldOption[];
```

Comment:
would this also account for empty options? 

Maybe unrelated to this representation but I was playing around with the api and options is only present on a dropdown type, so for ex this dropdown:
https://z3nmelsku-test.zendesk-staging.com/api/v2/ticket_fields/8722747260797
responds to an options call:
https://z3nmelsku-test.zendesk-staging.com/api/v2/ticket_fields/8722747260797/options

but a ticket field of another type, say a subject, returns"not found" when hitting the options endpoint
https://z3nmelsku-test.zendesk-staging.com/api/v2/ticket_fields/8504240744701/options

so im not sure thats a case we should also account for, test account [here](https://pandora-v2.zende.sk/resources/accounts/308999)

Reply:
This type is for the [ticket field object](https://developer.zendesk.com/api-reference/help_center/help-center-templates/objects/#ticket-field-object) returned in Curlybars as reported in the comment, so it is different from what we return in the APIs. Checking what we pass from the template, I can see that `options` can be `null` for most of the ticket fields.

In theory, a better type would be declaring a specific type for dropdowns and multiselect, where `options` is an array of `FieldOptions` and `null` for other types, but I didn't want to change too many things in this PR. I'll check if I can add the `| null` without causing the addition of too many checks around

Reply:
I tried to extend the type with `| null`, but it will require us to add a lot of null checks around. As mentioned, it would be better to define the type in a better way, where `options` is available or not based on the ticket field type. But I think that this can be eventually addressed later in a separate PR